extract/extract_tiktok.py

```diff
@@ -0,0 +1,230 @@
+            if tiktok_track["name"] == spotify_track["name"] or \
+                (fuzz.ratio(tiktok_track["name"].lower(), spotify_track["name"].lower())) > 90:
+                tiktok_track["in_spotify"] = True
+                spotify_track["in_tiktok"] = True
+                spotify_track["tiktok_rank"] = tiktok_track["tiktok_rank"]
+                break
+        if tiktok_track["in_spotify"] is False:
+            tiktok_not_on_spotify_chart.append(tiktok_track)
+    return tiktok_not_on_spotify_chart
+    
+
+def general_api_url_search(track_name: str, artist_names: list[str]) -> str:
+    '''
+    Returns a query url for spotify api by removing punctuation
+    '''
+    for i in track_name:
+        if i in string.punctuation:
+            track_name = track_name.split(i)
+            break
+    return f"{SPOTIFY_BASE_URL}search/?q=track:{track_name[0]} artist:{artist_names[0]}&type=track&limit=1"
+
+def search_api_track(track_name: str, artist_names: list[str], with_artist: bool, general_search: bool, headers):
+    '''
+    Searches the spotify API using a tiktok song name and artists' names
+    '''
+    if with_artist and general_search:
+        query_url = general_api_url_search(track_name, artist_names)
+    elif with_artist:
+        query_url = f"{SPOTIFY_BASE_URL}search/?q=track:{track_name} artist:{','.join(artist_names)}&type=track&limit=1"
+    else:
+        query_url = f"{SPOTIFY_BASE_URL}search/?q=track:{track_name}&type=track&limit=1"
+    print(query_url)
+    result = get(query_url, headers=headers)
+    try:
+        json_result = json.loads(result.content)["tracks"]["items"]
+    except:
+        json_result = json.loads(result.content)
+    return json_result
+
+
+def attempt_multiple_searches(song_name: str, song_artist: list[str], headers):
+    track = search_api_track(song_name, song_artist, True, False, headers)
+    if len(track) == 0:
+        print(f"Error obtaining track information for: {song_name}")
+        print("Attempting to find song in more general terms")
+        track = search_api_track(song_name, song_artist, True, True, headers)
+    if len(track) == 0:
+        print(f"Error obtaining track information for: {song_name} in more general terms")
+        print("Attempting to find song exclusively through song name")
+        track = search_api_track(song_name, song_artist, False, False,headers)
+    if len(track) == 0:
+        print(f"Cannot find information for track: {song_name}")
+        return None
+    if isinstance(track, dict):
+        if "error" in track.keys():
+            print(f"Cannot find song: {song_name} with artists: {','.join(song_artist)}")
+        return None
+    if "error" in track[0].keys():
+        print(f"Cannot find song: {song_name} with artists: {','.join(song_artist)}")
+        return None
+    return track
+
+
+
+def get_auth_token(client_id, client_secret) -> str:
+    """Gets an authorisation token from Spotify API"""
```

Comment:
Looks like we've set up the docstrings slightly different so it might be worth converting them all to your way just for the sake of consistency and neatness (i.e with single quotation marks and on separate lines to the quotes)